Client/Frontend/Browser/BrowserViewController/BrowserViewController+Wallet.swift

```diff
@@ -19,3 +73,175 @@ extension BrowserViewController: BraveWalletDelegate {
+            chainId != currentChainId else { return }
+      webView?.evaluateSafeJavaScript(
+        functionName: "window.ethereum.emit",
+        args: ["chainChanged", chainId],
+        contentWorld: .page,
+        completion: nil
+      )
+      updateEthereumProperties()
+    }
+  }
+  
+  func accountsChangedEvent(_ accounts: [String]) {
+    webView?.evaluateSafeJavaScript(
+      functionName: "window.ethereum.emit",
+      args: ["accountsChanged", accounts],
+      contentWorld: .page,
+      completion: nil
+    )
+    updateEthereumProperties()
+  }
+
+  @MainActor func allowedAccounts(_ includeAccountsWhenLocked: Bool) async -> ([String], BraveWallet.ProviderError, String) {
+    func filterAccounts(
+      _ accounts: [String],
+      selectedAccount: String?
+    ) -> [String] {
+      if let selectedAccount = selectedAccount, accounts.contains(selectedAccount) {
+        return [selectedAccount]
+      }
+      return accounts
+    }
+    // This method is called immediately upon creation of the wallet provider, which happens at tab
+    // configuration, which means it may not be selected or ready yet.
+    guard let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: false),
+          let origin = url?.origin?.asURL else {
+      return ([], .internalError, "Internal error")
+    }
+    let isLocked = await keyringService.isLocked()
+    if !includeAccountsWhenLocked && isLocked {
+      return ([], .success, "")
+    }
+    let selectedAccount = await keyringService.selectedAccount(.eth)
+    let permissions = Domain.ethereumPermissions(forUrl: origin)
+    return (
+      filterAccounts(permissions ?? [], selectedAccount: selectedAccount),
+      .success,
+      ""
+    )
+  }
+  
+  func updateEthereumProperties() {
+    Task { @MainActor in
+      /// Turn an optional value into a string (or quoted string in case of the value being a string) or
+      /// return `undefined`
+      func valueOrUndefined<T>(_ value: T?) -> String {
+        switch value {
+        case .some(let string as String):
+          return "\"\(string)\""
+        case .some(let value):
+          return "\(value)"
+        case .none:
+          return "undefined"
+        }
+      }
+      guard let webView = webView, let provider = walletProvider else {
+        return
+      }
+      let chainId = await provider.chainId()
+      webView.evaluateSafeJavaScript(
+        functionName: "window.ethereum.chainId = \"\(chainId)\"",
+        contentWorld: .page,
+        asFunction: false,
+        completion: nil
+      )
+      let networkVersion = valueOrUndefined(Int(chainId.removingHexPrefix, radix: 16))
+      webView.evaluateSafeJavaScript(
+        functionName: "window.ethereum.networkVersion = \(networkVersion)",
+        contentWorld: .page,
+        asFunction: false,
+        completion: nil
+      )
+      let selectedAccount = valueOrUndefined(await allowedAccounts(false).0)
```

Comment:
i'm little confused about the `allowedAccounts` which are from database and the `accounts` from `accountsChangedEvent`. Shouldnt we update eth properties with the `accounts` from `accountsChangedEvent`?

Reply:
Discussed in Slack, following desktops flow, which is just to fetch allowed accounts instead of taking whats in the events listener